File: Easy/JAVA/BranchSumsBT.java
import java.util.ArrayList;
import java.util.List;

public class BranchSumsBT {
    public static class Node {
        private int value;
        private Node left;
        private Node right;

        public Node(int value) {
            this.value = value;
        }
    }

    public class BTree {
        public static List<Integer> BrunchSum(Node node) {
            var sum = new ArrayList<Integer>();
            BrunchSum(node, 0, sum);
            return sum;
        }

        private static void BrunchSum(Node node, int currentSum, List<Integer> sum) {
            if (node == null)
                return;
            
            int endSum = currentSum + node.value;
            if (node.left == null && node.right == null) {
                sum.add(endSum);
                return;
            }

            BrunchSum(node.left, endSum, sum);
            BrunchSum(node.right, endSum, sum);
        }
    }
    public static void main(String[] args) {
        var root = new Node(10);
        root.left = new Node(5);
        root.left.left = new Node(2);
        root.left.left.left = new Node(1);
        root.left.right = new Node(5);
        root.right = new Node(15);
        root.right.left = new Node(13);
        root.right.left.right = new Node(14);
        root.right.right = new Node(22);

        var sums = BranchSumsBT.BTree.BrunchSum(root);
        System.out.println(sums.toString());
    }
}

File: Easy/PHP/BranchSumsBT.php
<?php

namespace Easy\PHP\BranchSumsBT;

class Node
{
    public $value;
    public $left;
    public $right;

    public function __construct($value)
    {
        $this->value = $value;
    }
}

class BTee
{
    public $root;

    public function __construct()
    {
        $this->root = null;
    }
    // Return the sum of nodes in binary tree
    public function branchSums($node)
    {
        if ($node == null) {
            return 0;
        }
        // left subtree sum +
        // right subtree sum +
        // current node sum
        return $this->branchSums($node->left) + $this->branchSums($node->right) + $node->value;
    }
}

$tree = new BTee();
$tree->root = new Node(10);
$tree->root->left = new Node(5);
$tree->root->left->left = new Node(2);
$tree->root->left->right = new Node(5);
$tree->root->right = new Node(11);
$tree->root->right->left = new Node(13);
$tree->root->right->left->right = new Node(14);

$result = $tree->branchSums($tree->root);
//var_dump("$result");
print_r("Branches sum: ". $result);
